core/unit_test/CMakeLists.txt

```diff
@@ -2,28 +2,29 @@
 # Add test-only library for gtest to be reused by all the subpackages
 #
 
+IF(NOT GTest_FOUND)  # fallback to internal gtest
+  SET(GTEST_SOURCE_DIR ${${PARENT_PACKAGE_NAME}_SOURCE_DIR}/tpls/gtest)
 
-SET(GTEST_SOURCE_DIR ${${PARENT_PACKAGE_NAME}_SOURCE_DIR}/tpls/gtest)
-
-#need here for tribits
-KOKKOS_INCLUDE_DIRECTORIES(${GTEST_SOURCE_DIR})
-KOKKOS_ADD_TEST_LIBRARY(
-  kokkos_gtest
-  HEADERS ${GTEST_SOURCE_DIR}/gtest/gtest.h
-  SOURCES ${GTEST_SOURCE_DIR}/gtest/gtest-all.cc
-)
+  #need here for tribits
+  KOKKOS_INCLUDE_DIRECTORIES(${GTEST_SOURCE_DIR})
+  KOKKOS_ADD_TEST_LIBRARY(
+    kokkos_gtest
+    HEADERS ${GTEST_SOURCE_DIR}/gtest/gtest.h
+    SOURCES ${GTEST_SOURCE_DIR}/gtest/gtest-all.cc
+  )
 
-# avoid deprecation warnings from MSVC
-TARGET_COMPILE_DEFINITIONS(kokkos_gtest PUBLIC GTEST_HAS_TR1_TUPLE=0 GTEST_HAS_PTHREAD=0)
+  # avoid deprecation warnings from MSVC
+  TARGET_COMPILE_DEFINITIONS(kokkos_gtest PUBLIC GTEST_HAS_TR1_TUPLE=0)
```

Comment:
This could be obsolete and the comment suggests that it's MSVC specific - lets see how the CI reacts.